source/shared/core_sqlsrv.h

```diff
@@ -1394,51 +1394,140 @@ struct param_meta_data
     SQLULEN get_column_size() { return column_size; }
 };
 
-// holds the output parameter information.  Strings also need the encoding and other information for
-// after processing.  Only integer, float, and strings are allowable output parameters.
-struct sqlsrv_output_param {
-
-    zval* param_z;
+// *** parameter struct used for SQLBindParameter ***
+struct sqlsrv_param
+{
+    SQLUSMALLINT    param_pos;      // 0-based - the position in the parameters of the statement
+    SQLSMALLINT     direction;
+    SQLSMALLINT     c_data_type;
+    SQLSMALLINT     sql_data_type;
+    SQLULEN         column_size;
+    SQLSMALLINT     decimal_digits;
+    SQLPOINTER      buffer;
+    SQLLEN          buffer_length;
+    SQLLEN          strlen_or_indptr;
+    int             param_php_type;
     SQLSRV_ENCODING encoding;
-    SQLUSMALLINT param_num;             // used to index into the ind_or_len of the statement
-    SQLLEN original_buffer_len;         // used to make sure the returned length didn't overflow the buffer
-    SQLSRV_PHPTYPE php_out_type;        // used to convert output param if necessary
-    bool is_bool;
-    param_meta_data meta_data;          // parameter meta data
+    bool            was_null;       // false by default - the original parameter was a NULL zval
+    zval            placeholder_z;  // A temp buffer for binding any input parameter, such as wide input string (UTF-16 buffer), the datetime strings, etc.
```

Comment:
Good point. I named it placeholder_z because it's also used in TVP as a placeholder for binding zval data post execution, so it's not just for input strings.

Reply:
If it's used post execution as well, is that for input parameters too?  Please update the comment to be clear

Reply:
yes only for input parameters